### PR TITLE
Add cached sizes key to cache, to ensure that invalidate_cache works correctly

### PR DESCRIPTION
**Pull request by rafiqhilali**
Fixes issue #172

### CODE REVIEW

File: avatar/forms.py [hunk 1]
```diff
@@ -49,8 +49,7 @@ def clean_avatar(self):
             })
 
         count = Avatar.objects.filter(user=self.user).count()
-        if (settings.AVATAR_MAX_AVATARS_PER_USER > 1 and
-                count >= settings.AVATAR_MAX_AVATARS_PER_USER):
+        if 1 < settings.AVATAR_MAX_AVATARS_PER_USER <= count:
```
Comment (rafiqhilali) [inline avatar/forms.py]: had to change this for linting to work

**Comment by johnyoonh** [comment]
It is now likely to do two two look ups and it also introduces another problem by not distinguishing different sizes. I think the original code had a flawed approach in using 'prefix'. I think it would be better to ditch 'prefix' in cache key and limit the caching to just avatar URLs. Currently when using `rebuild_avatars` command, the cached_funcs in avatar/utils.py is going to be an empty set and no cache is going to get deleted. I can introduce a PR, but it is going to break current implementation.

**Comment by johanneswilm** [comment]
Hey @johnyoonh, if you (or someone else) could write a PR for that, I can get it merged fairly quickly.

**Comment by johanneswilm** [comment]
Hey @rafiqhilali - any comment on @johnyoonh's comment here? I'd like to get something merged that (almost) everyone can agree on.

**Pull request merged**